File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        long length=s.size();
        string res="";
        string curr_res="";
        for(long i=0;i<length;i++){
            if(s[i]!=' '){
                curr_res+=s[i];
            }
            if(s[i]==' ' || i==length-1){
                if(!curr_res.empty()){
                    if(!res.empty()) res=" "+res;
                    res=curr_res+res;
                    curr_res="";
                }
            }
        }
        return res;
    }
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        String res="",curr_res="";
        for(int i=0;i<s.length();i++){
            if(s.charAt(i)!=' '){
                curr_res+=s.charAt(i);
            }
            if(s.charAt(i)==' ' || i==s.length()-1){
                if(!curr_res.equals("")){
                    if(!res.equals("")){
                        res=" "+res;
                    }
                    res=curr_res+res;
                    curr_res="";
                }
            }
        }
        return res;
    }
}